core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -52,10 +52,37 @@
 #include <HIP/Kokkos_HIP_Instance.hpp>
 #include <Kokkos_HIP.hpp>
 
+// TODO cannot use global variable on the device with ROCm 2.9
+//__device__ __constant__ unsigned long kokkos_impl_hip_constant_memory_buffer
+//    [Kokkos::Experimental::Impl::HIPTraits::ConstantMemoryUsage /
+//     sizeof(unsigned long)];
+
+namespace Kokkos {
+namespace Experimental {
+template <typename T>
+inline __device__ T *kokkos_impl_hip_shared_memory() {
+  extern __shared__ HIPSpace::size_type sh[];
+  return (T *)sh;
+}
+}  // namespace Experimental
+}  // namespace Kokkos
```

Comment:
Why do you close `Kokkos::Experimental::` to reopen next line?

Reply:
No good reason. It was done this way in CUDA.